pkg/applier/applier.go

```diff
@@ -254,7 +254,12 @@ func processApplyEvent(ctx context.Context, e event.ApplyEvent, s *stats.ApplyEv
 	}
 }
 
-func processWaitEvent(e event.WaitEvent, s *stats.WaitEventStats, objectStatusMap ObjectStatusMap) error {
+type waitEventProcessor struct {
```

Comment:
I don't love how this struct is new and unique for this method... other similar methods are on the supervisor directly, which won't work for this method, because it's specific to apply/destroy... 

What do you think about making an eventProcessor or eventHandler struct and putting all the handler methods on it? It would need to also hold the abandon method which needs the clientset. But it might make things a little cleaner. Then just make a new eventHandler every time apply or destroy is called.

Reply:
Fair point, done